### PR TITLE
Add Probe Functionality Support for amd platform.

### PR DESCRIPTION
**Pull request by reddysujith**
Add Probe Functionality Support for amd platform.

### CODE REVIEW

File: sound/soc/sof/amd/acp-ipc.c [hunk 1]
```diff
@@ -200,6 +202,22 @@ irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
 		return IRQ_HANDLED;
 	}
 
+	if (desc->probe_reg_offset) {
+		u32 val;
+		u64 posn;
+
+		val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, desc->probe_reg_offset);
+		if (val & BIT(31)) {
+			posn = val & ~BIT(31);
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-ipc.c]: maybe use a define for the magic 31, and explain what this test does?

Comment (plbossart) [inline sound/soc/sof/amd/acp-ipc.c]: why does posn need to be a 64-bit value...

File: sound/soc/sof/amd/acp-ipc.c [hunk 2]
```diff
@@ -200,6 +202,22 @@ irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
...
+			if (adata->probe_stream) {
+				adata->probe_stream->cstream_posn = posn;
+				snd_compr_fragment_elapsed(adata->probe_stream->cstream);
+				snd_sof_dsp_write(sdev, ACP_DSP_BAR, desc->probe_reg_offset, posn);
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-ipc.c:223]: ... when you write a 32-bit value here?

File: sound/soc/sof/amd/acp-probes.c [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	if (ret < 0) {
+		acp_dsp_stream_put(sdev, stream);
+		return ret;
+	}
+
+	/* write buffer size of stream in scratch memory */
+
+	buf_offset = sdev->debug_box.offset +
+		     offsetof(struct scratch_reg_conf, buf_size);
+	index = stream->stream_tag - 1;
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-probes.c:93]: is the stream_tag always > 0?

File: sound/soc/sof/amd/Kconfig [hunk 4]
```diff
@@ -21,6 +21,7 @@ config SND_SOC_SOF_AMD_COMMON
 	select SND_SOC_SOF_PCI_DEV
 	select SND_AMD_ACP_CONFIG
 	select SND_SOC_SOF_XTENSA
+	select SND_SOC_SOF_ACP_PROBES
```
Comment (plbossart) [inline sound/soc/sof/amd/Kconfig:24]: commit message: s/SoF/SOF

File: sound/soc/sof/amd/acp-ipc.c (hunk 1 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp-ipc.c]: done

Comment (reddysujith) [reply sound/soc/sof/amd/acp-ipc.c]: Done

File: sound/soc/sof/amd/acp-ipc.c (hunk 2 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp-ipc.c:223]: Done.

File: sound/soc/sof/amd/acp-probes.c (hunk 3 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp-probes.c:93]: Yes it is always greater than zero. 
 rechecked in the below link.
https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/amd/acp-stream.c#L183

**Comment by ranj063** [comment]
@reddysujith looks good. can you please fix the checkpatch warning?

**Review by bhiregoudar — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Review by bhiregoudar — DISMISSED** [review]

**Review by vijendarmukunda — DISMISSED** [review]

File: sound/soc/sof/amd/acp-ipc.c [hunk 5]
```diff
@@ -200,6 +202,27 @@ irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
...
+		/* Probe register consists of two parts
+		 * (0-30) bit has cumulative position value
+		 * 31 bit is a synchronization flag between DSP and CPU
+		 * for the position update
+		 */
+		val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, desc->probe_reg_offset);
+		if (val & PROBE_STATUS_BIT) {
+			posn = val & ~PROBE_STATUS_BIT;
+			if (adata->probe_stream) {
+				adata->probe_stream->cstream_posn = posn;
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-ipc.c:221]: should there be something that takes care of wrap-around if you use this capability for a long time the 32-bit byte counter will overflow?

**Review by plbossart — COMMENTED** [review]
looks mostly good but one more question below @reddysujith

File: sound/soc/sof/amd/acp-ipc.c (hunk 5 above)
Comment (reddysujith) [reply sound/soc/sof/amd/acp-ipc.c:221]: About this, We discussed internally , we almost store 2 GB,  For debugging it is ok, After that 31-bit overflow DSP will not send position update to host, Application won't read. There we can close the crecord command.

**Review by ranj063 — DISMISSED** [review]

File: sound/soc/sof/amd/acp-ipc.c (hunk 5 above)
Comment (plbossart) [reply sound/soc/sof/amd/acp-ipc.c:221]: ok, you probably want to add a comment here with the explanations above. Others will have the same comment when pushing this upstream.

Comment (reddysujith) [reply sound/soc/sof/amd/acp-ipc.c:221]: Done. Added comment and pushed.

**Review by plbossart — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Pull request merged**